src/object.rs

```diff
@@ -736,24 +738,33 @@ impl<T: IsA<Object> + SetValue> ObjectExt for T {
 
             for i in 0..details.n_params {
                 let arg_type = *(details.param_types.offset(i as isize)) & (!gobject_ffi::G_TYPE_FLAG_RESERVED_ID_BIT);
-                if arg_type != args[i as usize].type_().to_glib() {
+                if arg_type != args[i as usize].to_value_type().to_glib() {
                     return Err(BoolError("Incompatible argument types"));
                 }
             }
 
-            let mut c_args: Vec<gobject_ffi::GValue> = Vec::new();
-            let instance = Value::from(self);
-            c_args.push(ptr::read(instance.to_glib_none().0));
-            for arg in args {
-                c_args.push(ptr::read(arg.to_glib_none().0));
-            }
+            let mut v_args: Vec<Value>;
+            let mut s_args: [Value; 10] = mem::zeroed();
+            let args = if args.len() < 10 {
+                for (i, arg) in iter::once(&(self as &ToValue)).chain(args).enumerate() {
+                    s_args[i] = arg.to_value();
+                }
+                &s_args[0..args.len()+1]
+            } else {
+                v_args = Vec::with_capacity(args.len() + 1);
+                for arg in iter::once(&(self as &ToValue)).chain(args) {
```

Comment:
I'm not sure how to get rid of the `&(self as &ToValue)` disaster here. Any suggestions?

Reply:
Why it disaster?
By way, I proposed use `iter::once` for function, if used directly then old variant looks normal (after index fix and `with_capacity` on second branch)

Reply:
It looks ugly but works :) do you prefer it like this or the previous version?

Reply:
IMHO almost all same as long as it works

Reply:
👍 Let's keep it as now then